lib/cluster/index.ts

```diff
@@ -0,0 +1,663 @@
+import ClusterAllFailedError from '../errors/ClusterAllFailedError'
+import {defaults, noop} from '../utils/lodash'
+import ConnectionPool from './ConnectionPool'
+import {NodeKey, IRedisOptions, normalizeNodeOptions} from './util'
+
+var Deque = require('denque')
+var {isIP} = require('net')
+var Redis = require('../redis')
+var utils = require('../utils')
+var EventEmitter = require('events').EventEmitter
+var debug = require('../utils/debug')('ioredis:cluster')
+var ScanStream = require('../ScanStream').default
+var Commander = require('../commander')
+var Command = require('../command')
+var commands = require('redis-commands')
+var asCallback = require('standard-as-callback');
+var ClusterSubscriber = require('./ClusterSubscriber').default;
+var DelayQueue = require('./DelayQueue').default;
+var PromiseContainer = require('../promiseContainer');
+var {AbortError} = require('redis-errors');
+
+/**
+ * Creates a Redis Cluster instance
+ *
+ * @constructor
```

Comment:
You can probably omit most of those `@constructor` and other class related hints.